src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4006,6 +4006,36 @@ GenTree* Lowering::TryLowerAndOpToAndNot(GenTreeOp* andNode)
     return andnNode;
 }
 
+void Lowering::LowerBswapOp(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_BSWAP));
+
+    if (!comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE))
+    {
+        return;
+    }
+
+    GenTree* operand = node->gtGetOp1();
+    if (operand->OperIs(GT_IND))
+    {
+        operand->SetContained();
+
+        return;
```

Comment:
Just an observation: it is notable that this early return is significant as it prevent us from double-containing the `BSWAP` in a situation like `STOREIND(addr, BSWAP(IND))`.